boot/nuttx/include/sysflash/sysflash.h

```diff
@@ -28,8 +28,8 @@
 #define SECONDARY_ID    1
 #define SCRATCH_ID      2
 
-#define FLASH_AREA_IMAGE_PRIMARY(x)    PRIMARY_ID
-#define FLASH_AREA_IMAGE_SECONDARY(x)  SECONDARY_ID
+#define FLASH_AREA_IMAGE_PRIMARY(x)    ((x == 0) ? PRIMARY_ID : PRIMARY_ID)
+#define FLASH_AREA_IMAGE_SECONDARY(x)  ((x == 0) ? SECONDARY_ID : SECONDARY_ID)
```

Comment:
It would be better to enclose the `x` usage with parenthesis on these macros.

Reply:
You're right, just fixed it.